lib/Dialect/SV/Transforms/GeneratorCallout.cpp

```diff
@@ -106,35 +92,58 @@ void RTLGeneratorCalloutPass::processGenerator(
   // Assumption: All the options required by the generator program must be
   // present in the schema.
   for (auto attr : genSchema.requiredAttrs()) {
-    // Get the port name from schema.
-    StringRef portName = attr.dyn_cast<StringAttr>().getValue();
-    generatorArgs.push_back("--" + portName.str());
-    // Get the value for the corresponding port name.
-    // Assumption: the value always exists.
-    auto v = generatedModuleOp->getAttr(portName);
-    if (auto intV = v.dyn_cast<IntegerAttr>())
-      generatorArgs.push_back(intV.getValue().toString(10, false));
-    else if (auto strV = v.dyn_cast<StringAttr>())
-      generatorArgs.push_back(strV.getValue().str());
+    if (auto sAttr = attr.dyn_cast<StringAttr>()) {
```

Comment:
Is the "non-stringattr case an error or should it just be ignored like this?

I'd recommend inverting this control flow to de-nest it a level, something like:

```
auto sAttr = attr.dyn_cast...
if (!sAttr) continue;
```